File: internal/escpos/Finish.go
package escpos

import "bytes"

// FeedBytes 返回 n 个换行(走纸 n 行);n<=0 返回空。不含初始化。
func FeedBytes(n int) []byte {
	if n <= 0 {
		return nil
	}
	return bytes.Repeat([]byte{0x0A}, n)
}

// Finish 生成打印收尾字节:尾部走纸(TailFeed=按纸宽的基数+偏移)+ 可选半切(GS V 1)。
// 不含初始化;由 printsvc 统一追加到内容之后,保证切纸前走纸足够清过切刀。
func Finish(widthMM, tailOffset int, cut bool) []byte {
	out := FeedBytes(TailFeed(widthMM, tailOffset))
	if cut {
		out = append(out, 0x1D, 0x56, 0x01)
	}
	return out
}

File: Android/app/src/main/java/com/congmingpay/android/escpos/Finish.kt
package com.congmingpay.android.escpos

/**
 * 打印收尾字节生成。
 */
object Finish {
    /** 返回 n 个换行（走纸 n 行）；n<=0 返回空 */
    fun feedBytes(n: Int): ByteArray {
        if (n <= 0) return ByteArray(0)
        return ByteArray(n) { 0x0A }
    }

    /**
     * 生成打印收尾字节：尾部走纸（TailFeed=按纸宽的基数+偏移）+ 可选半切（GS V 1）。
     * 不含初始化；由 PrintDispatcher 统一追加到内容之后。
     */
    fun finish(widthMM: Int, tailOffset: Int, cut: Boolean): ByteArray {
        val out = java.io.ByteArrayOutputStream()
        out.write(feedBytes(Layout.tailFeed(widthMM, tailOffset)))
        if (cut) {
            out.write(0x1D)
            out.write(0x56)
            out.write(0x01)
        }
        return out.toByteArray()
    }
}
